src/gateways/RedisCachedRegistrationGateway.ts

```diff
@@ -0,0 +1,29 @@
+import { FormCacheFactory } from "../lib/formCache";
+import { Registration } from "../lib/registration/registration";
+import { IRegistration } from "../lib/registration/types";
+import { CachedRegistrationGateway } from "./CachedRegistrationGateway";
+
+export class RedisCachedRegistrationGateway
```

Comment:
Gone with the `<Implementation><Abstraction>` naming pattern.  @tm1303 and @Ninamma and I discussed this a lil bit.

Let's not go with `CachedRegistrationGatewayImpl` because it missed the opportunity to be more descriptive.  This way we can easily swap out `RedisCachedRegistrationGateway` for `MemcachedCachedRegistrationGateway` etc.

@tm1303 expressed a preference for something like `CachedRegistrationRedisGateway`.  To my eyes when parsing text this looks like `<Abstr><Implementation><action>` 😅 But I'm not too strongly opinionated.